src/sieve.py

```diff
@@ -16,5 +19,13 @@ def sieve(n: int) -> list[int]:
     primes = []
 
     # FIXME: fill out this bit
-
+    while len(candidates)>0:
```

Comment:
Lists will evaluate to `False` when you test them, so this could be simplified to `while candidates`.